crates/orchestrator/src/jobs/mod.rs

```diff
@@ -124,34 +126,69 @@ pub trait Job: Send + Sync {
 }
 
 /// Creates the job in the DB in the created state and adds it to the process queue
+#[tracing::instrument(name = "general_create_job", skip(config))]
 pub async fn create_job(
     job_type: JobType,
     internal_id: String,
     metadata: HashMap<String, String>,
     config: Arc<Config>,
 ) -> Result<(), JobError> {
+    let block_gauge = telemetry::global_meter()
+        .f64_gauge(format!("{:?}{}", OTEL_SERVICE_NAME, "_block_state"))
+        .with_description("A gauge to show block state at given time")
+        .with_unit("block")
+        .init();
+
+    tracing::info!("CREATE JOB CALLED for {:?} and type {:?}", internal_id, job_type);
+
     let existing_job = config
         .database()
         .get_job_by_internal_id_and_type(internal_id.as_str(), &job_type)
         .await
         .map_err(|e| JobError::Other(OtherError(e)))?;
+
     if existing_job.is_some() {
+        tracing::info!("CREATE JOB SKIPPED for {:?} and type {:?}", internal_id, job_type);
         return Err(JobError::JobAlreadyExists { internal_id, job_type });
     }
 
     let job_handler = factory::get_job_handler(&job_type).await;
-    let job_item = job_handler.create_job(config.clone(), internal_id, metadata).await?;
+    let job_item = job_handler.create_job(config.clone(), internal_id.clone(), metadata).await?;
     config.database().create_job(job_item.clone()).await.map_err(|e| JobError::Other(OtherError(e)))?;
 
     add_job_to_process_queue(job_item.id, config.clone()).await.map_err(|e| JobError::Other(OtherError(e)))?;
+
+    tracing::info!("JOB CREATED for {:?} and type {:?} with Id {:?}", internal_id, job_type, job_item.id);
+
+    let attributes = [
+        KeyValue::new("job_type", format!("{:?}", job_type)),
```

Comment:
can we not just to `.to_string()`

Reply:
These types do not implement `to_string()` method, 
Do we wanna implement that or continue with Debug.
WDYT @apoorvsadana ?